noxfile.py

```diff
@@ -0,0 +1,17 @@
+from __future__ import annotations
+
+import nox  # type: ignore
+
+
+@nox.session(python=["3.8", "3.9", "3.10", "3.11"])
+def tests(session: nox.Session) -> None:
+    session.install("-r", "rerun_py/requirements-build.txt")
+    session.install("./rerun_py")
+    session.run("just", "py-test")
+
+
+@nox.session(python=["3.8", "3.9", "3.10", "3.11"])
+def examples(session: nox.Session) -> None:
+    session.install("-r", "examples/python/requirements.txt")
+    session.install("./rerun_py")
+    session.run("python", "scripts/run_all.py", "--save")
```

Comment:
could this file be moved to e.g. scripts/ so as not to pollute the repo root further?

Reply:
Yes, but that would mean running `nox -f scripts/noxfile.py -s tests`. I guess that can be hidden behind `just nox -s tests`. Prefer that way?

Reply:
I just looked into it, and I'm a bit split. It's not tooo bad to move it and force using `just nox ...`, but does generate some friction wrt usual nox expectations, incl:
- one must change the current directory to `..` in each session function
- the `.nox` cache dir weirdly ends up in `scripts/` as well

Also, I believe nox could help with #2200, so I'm inclined to give it first-class citizen status.

Reply:
ok!